tests/test_data_processor.py

```diff
@@ -499,11 +555,58 @@ def test_max_altitude(self, data_processor):
             ),
         ],
     )
-    def test_calculate_rotations(self, csv_path, expected_value, n_packets):
+    def test_calculate_rotations_quaternions(self, csv_path, expected_value, n_packets):
         data_packets = load_data_packets(csv_path, n_packets)
         d = IMUDataProcessor()
         d.update(data_packets)
         rotations = d._rotated_accelerations
         assert len(rotations) == n_packets
 
         assert rotations[-1] == pytest.approx(expected_value)
+
+    @pytest.mark.parametrize(
+        ("csv_path", "expected_value", "n_packets"),
+        [
+            (
+                "tests/imu_data/xminus.csv",
+                9.85116094,
+                2,
+            ),
+            (
+                "tests/imu_data/yminus.csv",
+                9.83891064,
+                2,
+            ),
+            (
+                "tests/imu_data/zminus.csv",
+                9.82264007,
+                2,
+            ),
+            (
+                "tests/imu_data/xplus.csv",
+                9.75015129,
+                2,
+            ),
+            (
+                "tests/imu_data/yplus.csv",
+                9.61564675,
+                2,
+            ),
+            (
+                "tests/imu_data/zplus.csv",
+                9.81399729,
+                2,
+            ),
+        ],
+    )
+    def test_calculate_rotations_gyro(self, csv_path, expected_value, n_packets):
+        data_packets = load_data_packets(csv_path, n_packets)
+        d = IMUDataProcessor()
+        for i, _ in enumerate(data_packets):
+            data_packets[i].estAttitudeUncertQuaternionW = 1
+        print(data_packets)
+        d.update(data_packets)
+        rotations = d._rotated_accelerations
+        assert len(rotations) == n_packets
+
+        assert rotations[-1] == pytest.approx(expected_value)
```

Comment:
Interesting that the expected_value is the same for both methods. I would've expected a small difference at least.